api/src/main/kotlin/com/playkids/onboarding/api/route/ProfileRoutes.kt

```diff
@@ -28,10 +35,12 @@ fun Route.profileRouting(profileService: ProfileService) {
 
             call.respond(profile)
         }
-        post<Profile> { profile ->
-            profileService.create(profile)
+        post<CreateProfileDTO> { profileDTO ->
+            val profile = profileService.create(profileDTO)
 
-            call.respondText("Profile Created", status = HttpStatusCode.OK)
+            sqsHandler.handleInsert(profile)
```

Comment:
I have a doubt about this. If i call sqsHandler inside de service I would have to either import the :api project inside the core or declare de sqsHandler inside the core. Either way I would be delegating responsibility about emitting sqs events to the core. My question is if this is a good thing, since I import the core project in other projects, like the data-listener (through postgres project). So it's okay for me to do this? At first I was thinking about not mixing the sender and receiver projects. Or should I create and ProfileApiService in the api project, and from there call the emitter and the ProfileService?

Reply:
Answered in Discord :)